rust/core-lib/src/plugins/catalog.rs

```diff
@@ -48,6 +48,11 @@ impl<'a> PluginCatalog {
     pub fn reload_from_paths(&mut self, paths: &[PathBuf]) {
         self.items.clear();
         self.locations.clear();
+        self.load_from_paths(paths);
+    }
+
+    /// Loads plugins from paths and adds them to existing plugins.
+    pub fn load_from_paths(&mut self, paths: &[PathBuf]) {
```

Comment:
I prefer `pub(crate)` for everything where possible. A lot of this code predates `pub(crate)` existing, so can be a bad example.

The main reason `pub(crate)` is better is that it lets the compiler give us more unused-code warnings. If something is marked `pub`, it doesn't warn us when it's unused because it might be used by another crate.